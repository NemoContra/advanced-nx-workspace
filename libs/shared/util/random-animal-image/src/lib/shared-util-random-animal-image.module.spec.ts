import * as Module from './shared-util-random-animal-image.module';
import { SharedUtilRandomAnimalImageModule } from './shared-util-random-animal-image.module';
import { createServiceFactory } from '@ngneat/spectator/jest';
import {
  ANIMAL_IMAGE_CONFIG,
  AnimalImageConfig,
  SharedUtilRandomAnimalImageService,
} from '@flight-workspace/shared/util/random-animal-image';

jest.spyOn(Module, 'configFactory');

console.log = jest.fn();

describe('SharedUtilRandomAnimalImageModule', () => {
  describe.each([
    ['forRoot', { cats: true, dogs: true }],
    ['forRoot', { cats: true, dogs: false }],
    ['forRoot', { cats: false, dogs: true }],
    ['forRoot', { cats: false, dogs: false }],
    ['forChild', { cats: true, dogs: true }],
    ['forChild', { cats: true, dogs: false }],
    ['forChild', { cats: false, dogs: true }],
    ['forChild', { cats: false, dogs: false }],
  ] as const)(
    'call %s with %j',
    (registerFn: 'forRoot' | 'forChild', config: AnimalImageConfig) => {
      const createService = createServiceFactory({
        service: SharedUtilRandomAnimalImageService,
        imports: [SharedUtilRandomAnimalImageModule[registerFn](config)],
      });

      it('should register', () => {
        const spectator = createService();
        expect(spectator.inject(ANIMAL_IMAGE_CONFIG)).toMatchSnapshot('config');
        expect(Module.configFactory).toHaveBeenCalledWith(config);
      });
    }
  );

  test('configFactory', () => expect(Module.configFactory).toMatchSnapshot());
});
