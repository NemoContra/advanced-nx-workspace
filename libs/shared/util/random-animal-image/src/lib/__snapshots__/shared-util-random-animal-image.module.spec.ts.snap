// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedUtilRandomAnimalImageModule call forChild with {"cats":false,"dogs":false} should register: config 1`] = `
Object {
  "cats": false,
  "dogs": false,
}
`;

exports[`SharedUtilRandomAnimalImageModule call forChild with {"cats":false,"dogs":true} should register: config 1`] = `
Object {
  "cats": false,
  "dogs": true,
}
`;

exports[`SharedUtilRandomAnimalImageModule call forChild with {"cats":true,"dogs":false} should register: config 1`] = `
Object {
  "cats": true,
  "dogs": false,
}
`;

exports[`SharedUtilRandomAnimalImageModule call forChild with {"cats":true,"dogs":true} should register: config 1`] = `
Object {
  "cats": true,
  "dogs": true,
}
`;

exports[`SharedUtilRandomAnimalImageModule call forRoot with {"cats":false,"dogs":false} should register: config 1`] = `
Object {
  "cats": false,
  "dogs": false,
}
`;

exports[`SharedUtilRandomAnimalImageModule call forRoot with {"cats":false,"dogs":true} should register: config 1`] = `
Object {
  "cats": false,
  "dogs": true,
}
`;

exports[`SharedUtilRandomAnimalImageModule call forRoot with {"cats":true,"dogs":false} should register: config 1`] = `
Object {
  "cats": true,
  "dogs": false,
}
`;

exports[`SharedUtilRandomAnimalImageModule call forRoot with {"cats":true,"dogs":true} should register: config 1`] = `
Object {
  "cats": true,
  "dogs": true,
}
`;

exports[`SharedUtilRandomAnimalImageModule configFactory 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "cats": true,
        "dogs": true,
      },
    ],
    Array [
      Object {
        "cats": true,
        "dogs": false,
      },
    ],
    Array [
      Object {
        "cats": false,
        "dogs": true,
      },
    ],
    Array [
      Object {
        "cats": false,
        "dogs": false,
      },
    ],
    Array [
      Object {
        "cats": true,
        "dogs": true,
      },
    ],
    Array [
      Object {
        "cats": true,
        "dogs": false,
      },
    ],
    Array [
      Object {
        "cats": false,
        "dogs": true,
      },
    ],
    Array [
      Object {
        "cats": false,
        "dogs": false,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;
